src/librustc/ty/mod.rs

```diff
@@ -637,8 +638,7 @@ impl<'tcx> RegionParameterDef<'tcx> {
     }
 
     pub fn to_bound_region(&self) -> ty::BoundRegion {
-        // this is an early bound region, so unaffected by #32330
-        ty::BoundRegion::BrNamed(self.def_id, self.name, Issue32330::WontChange)
+        ty::BoundRegion::BrNamed(self.def_id, self.name, self.issue_32330)
```

Comment:
So we're having that big enum on every `ty::Region`? I suppose we should intern the `(DefId, Name, Issue32230)` triple and check the perf impact.